File: APOD.py
import requests

def get_apod(api_key=None):
    base_url = "https://api.nasa.gov/planetary/apod"

    # If you have an API key, you can include it in the request
    if api_key:
        params = {'api_key': api_key}
    else:
        params = {}

    response = requests.get(base_url, params=params)

    if response.status_code == 200:
        data = response.json()
        return data
    else:
        print(f"Error: {response.status_code}")
        return None


# Replace 'YOUR_API_KEY' with your actual NASA API key
nasa_api_key = 'YOUR_API_Key'
apod_data = get_apod(api_key=nasa_api_key)

if apod_data:
    print("Title:", apod_data['title'])
    print("Date:", apod_data['date'])
    print("Explanation:", apod_data['explanation'])
    print("HD URL:", apod_data['hdurl'])
    print("URL:", apod_data['url'])
else:
    print("Failed to retrieve APOD data.")

File: test/APOD.js
const nasaApiKey = process.env.APOD_KEY;
let apodTitle, apodDate, apodExplanation, apodHdUrl, apodUrl;

async function getApod(apiKey) {
    const baseUrl = "https://api.nasa.gov/planetary/apod";
    
    // If you have an API key, include it in the request
    const params = apiKey ? { api_key: apiKey } : {};
    
    const response = await fetch(`${baseUrl}?${new URLSearchParams(params)}`);
    
    if (response.ok) {
        const data = await response.json();
        return data;
    } else {
        console.error(`Error: ${response.status}`);
        return null;
    }
}


getApod(nasaApiKey)
    .then(apodData => {
        if (apodData) {
            // Assign values to variables
            apodTitle = apodData.title;
            apodDate = apodData.date;
            apodExplanation = apodData.explanation;
            apodHdUrl = apodData.hdurl;
            apodUrl = apodData.url;

            // Manipulate the HTML DOM to display the values
            document.getElementById('title').innerText = `Title: ${apodTitle}`;
            document.getElementById('date').innerText = `Date: ${apodDate}`;
            document.getElementById('explanation').innerText = `Explanation: ${apodExplanation}`;
            document.getElementById('hdUrl').innerText = `HD URL: ${apodHdUrl}`;
            document.getElementById('url').innerText = `URL: ${apodUrl}`;
        } else {
            console.log("Failed to retrieve APOD data.");
        }
    })
    .catch(error => console.error("Error:", error));
